jaeger-core/src/main/java/com/uber/jaeger/baggage/RemoteBaggageRestrictionManager.java

```diff
@@ -0,0 +1,141 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.uber.jaeger.baggage;
+
+import com.uber.jaeger.baggage.http.BaggageRestrictionResponse;
+import com.uber.jaeger.exceptions.BaggageRestrictionManagerException;
+import com.uber.jaeger.metrics.Metrics;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Timer;
+import java.util.TimerTask;
+
+/**
+ * NewRestrictionManager returns a BaggageRestrictionManager that polls the agent for the latest
+ * baggage restrictions.
+ */
+public class RemoteBaggageRestrictionManager implements BaggageRestrictionManager {
+  private static final int DEFAULT_REFRESH_INTERVAL_MS = 60000;
+  private static final int DEFAULT_INITIAL_DELAY_MS = 0;
+
+  private final String serviceName;
+  private final BaggageRestrictionManagerProxy proxy;
+  private final Timer pollTimer;
+  private final Metrics metrics;
+  private final boolean denyBaggageOnInitializationFailure;
+  private volatile boolean initialized;
+  private volatile Map<String, Restriction> restrictions = new HashMap<String, Restriction>();
+  private final Restriction invalidRestriction;
+  private final Restriction validRestriction;
+
+  public RemoteBaggageRestrictionManager(
+      String serviceName, BaggageRestrictionManagerProxy proxy, Metrics metrics, boolean denyBaggageOnInitializationFailure) {
+    this(serviceName, proxy, metrics, denyBaggageOnInitializationFailure, DEFAULT_REFRESH_INTERVAL_MS);
+  }
+
+  public RemoteBaggageRestrictionManager(
+      String serviceName,
+      BaggageRestrictionManagerProxy proxy,
+      Metrics metrics,
+      boolean denyBaggageOnInitializationFailure,
+      int refreshIntervalMs
+  ) {
+    this(serviceName, proxy, metrics, denyBaggageOnInitializationFailure, refreshIntervalMs, DEFAULT_INITIAL_DELAY_MS);
+  }
+
+  RemoteBaggageRestrictionManager(
+      String serviceName,
+      BaggageRestrictionManagerProxy proxy,
+      Metrics metrics,
+      boolean denyBaggageOnInitializationFailure,
+      int refreshIntervalMs,
+      int initialDelayMs
```

Comment:
initialDelayMs is only exposed for testing purposes so I can determine when the first call to remote agent is made. Under normal operations, this class will start up and asynchronously fetch restrictions. If the user wants to know if restrictions are ready, they can check via isReady()